src/components/Login/index.js

```diff
@@ -0,0 +1,67 @@
+import styled from "styled-components";
+import PropTypes from "prop-types";
+import isDefined from "@codewell/is-defined";
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+export const Login = ({ onChange, endpoint, ...props }) => {
+  const [value, setValue] = useState({});
+
+  const setError = (error) => {
+    setValue({
+      error: error,
+    });
+  };
+
+  useEffect(() => {
+    axios.get(`${endpoint}/user`)
+      .then((result) => {
+        setValue({
+          loggedIn: true,
+          user: result.data,
+        })
+      })
+      .catch((error) => {
+        if (error.response) {
+          switch (error.response.status) {
+            case 401:
+              return setValue({loggedIn: false});
+            default:
+              return setError(`${error.response.status} on user response`);
+          }
+        } else if (error.request) {
+          setError(`no response from user request: ${error.message}`);
+        } else {
+          setError(`user request not sent: ${error.message}`);
+        }
+      })
+  }, [axios]);
+
+  useEffect(() => {
```

Comment:
This should definetely be a hook that we build later on that we can just import! Super simple but very useful. Like 
```javascript
useCallback(state, callback) // Trigger callback on state change
```